tests/02-list_valid_components.bats

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bats
+
+load libs/shared_setup
+
+setup () {
+    create_git_branch MOODLE_31_STABLE v3.1.0
+
+    cd $BATS_TEST_DIRNAME/../list_valid_components/
+}
+
+@test "list_valid_components: 31 components" {
+    # We need to rewrite the directory in the fixture for verifying correct result.
+    EXPECTED=$WORKSPACE/valid_components_expected.txt
+    sed -e "s#\[gitdir\]#${gitdir}#" $BATS_TEST_DIRNAME/fixtures/31-valid_components.txt > $EXPECTED
+
+    OUTPUT=$WORKSPACE/valid_components_out.txt
+
+    # Run test
+    php list_valid_components.php --basedir=$gitdir > $OUTPUT
+    assert_success
+    diff -ruN $EXPECTED $OUTPUT
+    assert_output ''
+
+    rm $EXPECTED $OUTPUT
```

Comment:
@stronk7 
1) Yeah you are probably right. When i first wrote the test I wasn't using the workspace directory
2) [It does](https://github.com/sstephenson/bats#setup-and-teardown-pre--and-post-test-hooks), though I am not sure we want to clean up on every execution. I do clean up the workspace dir [on the start of each test run ](https://github.com/moodlehq/moodle-local_ci/blob/c523b79/tests/libs/shared_setup.bash#L69) - but I did this because I found there were situations when I killed the tests and then the teardown type operations were not run (well, understandably).